### PR TITLE
#6: NullReferenceException in Bugsnag.NET.Request.StackTraceLine.<Build>d__1.MoveNext()

### PR DESCRIPTION
**Pull request by IgorBurakov**
Fixed cases when either Stack or TargetSite is null in the Exception object

### CODE REVIEW

**Comment by IgorBurakov** [comment]
It's appeared that if at least one of the Exceptions in the Event doesn't have any StackTrace line, then Bugsnag server doesn't create a record about the event, although the API call returns OK.

Thus I'm creating an empty line now for the StackTrace if it is absent in the original Exception

File: Bugsnag.NET.Tests/Request/NoticeTests.cs [hunk 1]
```diff
@@ -24,6 +25,17 @@ public void HasCorrectSchema()
...
 
+        [Test]
+        public void StackIsAbsent()
+        {
+            const string InternalErrorMessage = "An internal error has occurred!";
+            Exception ex = new Exception(InternalErrorMessage, new Exception("Debug Info: " + Environment.NewLine + "Some debug information goes here..."));
+            var ev = new Event(ex);
+            var notice = new Notice("apikey", new Notifier(), ev);
+            var json = JsonConvert.SerializeObject(notice);
+            // should reach the end of function without any exception thrown
```
Comment (awseward) [inline Bugsnag.NET.Tests/Request/NoticeTests.cs]: Would you mind wrapping the appropriate lines here in an `Assert.DoesNotThrow` call?

File: Bugsnag.NET/Request/Event.cs [hunk 2]
```diff
@@ -93,10 +93,12 @@ static string _GetGroupingHash(IEnumerable<Exception> unwrapped)
         static string _GetContext(Exception ex)
         {
             if (ex == null) { return string.Empty; }
+            var targetSite = ex.TargetSite;
 
-            return string.Format("{0}::{1}",
-                ex.TargetSite.DeclaringType,
-                ex.TargetSite.Name);
+            return targetSite == null ? "NullTargetSite" :
```
Comment (awseward) [inline Bugsnag.NET/Request/Event.cs]: Any chance you wouldn't mind using the string `null::null` for this situation?

**Comment by awseward** [comment]
@IgorBurakov: Sorry about the slow response time on this! I'd love to get this merged in, but just have a couple of minor comments before doing so.

File: Bugsnag.NET/Request/Event.cs (hunk 2 above)
Comment (IgorBurakov) [inline Bugsnag.NET/Request/Event.cs]: Sure, nice catch!

File: Bugsnag.NET.Tests/Request/NoticeTests.cs (hunk 1 above)
Comment (IgorBurakov) [inline Bugsnag.NET.Tests/Request/NoticeTests.cs]: The test method result (success or fail) is the same in both cases.
According to the http://www.mikedellanoce.com/2010/12/assertdoesnotthrow-why-does-it-exist.html using Assert.DoesNotThrow hides the details of original exception and makes it harder to get the reason of the error.

Comment (awseward) [inline Bugsnag.NET.Tests/Request/NoticeTests.cs]: Interesting! Sounds good!

**Comment by awseward** [comment]
Thanks, Igor!
Hoping to push out a new nupkg today!

**Comment by IgorBurakov** [comment]
Sounds good, Andrew! It was a pleasure for me to put my 2 cents into the great lib for everyone's benefit :)

**Pull request merged**